views.py

```diff
@@ -40,10 +45,48 @@ def landing(request):
 
 def display(request, pid):
     obj = repo.get_object(pid, create=False)
+    if not obj.exists:
+        raise Http404
+    template_info = {'obj': obj}
+    content_models = obj.get_models()
+    if URIRef('info:fedora/bdr-cmodel:implicit-set') in content_models:
+        template_info['obj_type'] = 'implicit-set'
+    else:
+        template_info['obj_type'] = ''
     return render(
         request,
         template_name='repo_direct/display.html',
-        dictionary={'obj': obj}
+        dictionary=template_info,
+    )
+
+
+@login_required
+def reorder(request, pid):
+    form = ReorderForm(request.POST or None)
+    if request.method == 'POST':
+        if form.is_valid():
+            child_pids_ordered_list = form.cleaned_data['child_pids_ordered_list'].split(u',')
+            pairs_param_for_api = json.dumps([(value, str(index+1)) for index, value in enumerate(child_pids_ordered_list)])
+            r = requests.post(settings.REORDER_URL, data={'pairs': pairs_param_for_api})
+            if r.ok:
+                messages.info(request, 'New order has been submitted (allow a bit of time for the changes to appear)')
+                return HttpResponseRedirect(reverse('repo_direct:display', args=(pid,)))
+            else:
+                raise Exception('error submitting new order')
+    bdr_item = bdrcommon.BdrItem(pid, bdr_server, identities=[settings.BDR_ADMIN])
```

Comment:
Nonblocking for this PR, but a note for the future.  When we get a chance, we should refactor into the models file.  And add any functionality to convert child objects/parts within the model.